tests/phpunit/tests/rest-api/rest-global-styles-revisions-controller.php

```diff
@@ -0,0 +1,252 @@
+		);
+
+		// Checks that the revisions are returned for all eligible users.
+		wp_set_current_user( self::$second_admin_id );
+		$config['styles']['color']['background'] = 'blue';
+		$new_styles_post                         = array(
+			'ID'           => self::$global_styles_id,
+			'post_content' => wp_json_encode( $config ),
+		);
+
+		wp_update_post( $new_styles_post, true, false );
+
+		$request  = new WP_REST_Request( 'GET', '/wp/v2/global-styles/' . self::$global_styles_id . '/revisions' );
+		$response = rest_get_server()->dispatch( $request );
+		$data     = $response->get_data();
+
+		$this->assertCount( 2, $data, 'Check that two revisions exists' );
+		$this->assertEquals( self::$second_admin_id, $data[0]['author'], 'Check that second author id returns expected value' );
+		$this->assertEquals( self::$admin_id, $data[1]['author'], 'Check that second author id returns expected value' );
+	}
+
+	/**
+	 * @covers WP_REST_Global_Styles_Controller::get_item_schema
+	 *
+	 * @ticket 58524
+	 */
+	public function test_get_item_schema() {
+		$request    = new WP_REST_Request( 'OPTIONS', '/wp/v2/global-styles/' . self::$global_styles_id . '/revisions' );
+		$response   = rest_get_server()->dispatch( $request );
+		$data       = $response->get_data();
+		$properties = $data['schema']['properties'];
+
+		$this->assertCount( 9, $properties, 'Schema properties array does not have exactly 9 elements' );
+		$this->assertArrayHasKey( 'id', $properties, 'Schema properties array does not have "id" key' );
+		$this->assertArrayHasKey( 'styles', $properties, 'Schema properties array does not have "styles" key' );
+		$this->assertArrayHasKey( 'settings', $properties, 'Schema properties array does not have "settings" key' );
+		$this->assertArrayHasKey( 'parent', $properties, 'Schema properties array does not have "parent" key' );
+		$this->assertArrayHasKey( 'author', $properties, 'Schema properties array does not have "author" key' );
+		$this->assertArrayHasKey( 'date', $properties, 'Schema properties array does not have "date" key' );
+		$this->assertArrayHasKey( 'date_gmt', $properties, 'Schema properties array does not have "date_gmt" key' );
+		$this->assertArrayHasKey( 'modified', $properties, 'Schema properties array does not have "modified" key' );
+		$this->assertArrayHasKey( 'modified_gmt', $properties, 'Schema properties array does not have "modified_gmt" key' );
+	}
+
+	/**
+	 * @covers WP_REST_Global_Styles_Controller::get_item_permissions_check
+	 *
+	 * @ticket 58524
+	 */
+	public function test_get_item_permissions_check() {
+		wp_set_current_user( self::$author_id );
+		$request  = new WP_REST_Request( 'GET', '/wp/v2/global-styles/' . self::$global_styles_id . '/revisions' );
+		$response = rest_get_server()->dispatch( $request );
+
+		$this->assertErrorResponse( 'rest_cannot_view', $response, 403 );
+	}
+
+	/**
+	 * @doesNotPerformAssertions
+	 */
+	public function test_context_param() {
+		// Controller does not use get_context_param().
```

Comment:
The controller does have `context` set, so this should get tests.

Reply:
Thanks for that.

I was looking at how other tests implement `test_context_param()` ([example](https://github.com/WordPress/wordpress-develop/blob/trunk/tests/phpunit/tests/rest-api/rest-categories-controller.php#L113)), and they're testing the data in `$data['endpoints']` after an `OPTIONS` call. 

This controller doesn't have any other endpoint, e.g., `WP_REST_Server::CREATABLE` and so I'm not sure how to test it.  `$data['endpoints']` doesn't have any testable data. Any tips?

I'll add another `get_items` test to check that the correct fields are returned however.

Reply:
I've added a test `test_get_item_embed_context` to check the difference when sending context as an arg in the GET request